packages/odyssey-react-mui/src/Button.tsx

```diff
@@ -167,10 +193,15 @@ const Button = ({
   variant: variantProp,
 }: ButtonProps) => {
   const muiProps = useMuiProps();
+  const background = useBackground();
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+  // Determine if the button is in a gray background and is a secondary variant
+  const isGrayBackground = background === "gray" && variantProp === "secondary";
```

Comment:
I wanna get away from `gray`. `isLowContrastBackground` or something more generic that we could use.